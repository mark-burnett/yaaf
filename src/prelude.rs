pub use crate::{
    actor::{Actor, Tell},
    context::{Context, Publish},
    handler::Handler,
    source::Source,
    system::System,
};
pub use ::async_trait::async_trait;
pub use ::yaaf_macros::{Actor, Source};
