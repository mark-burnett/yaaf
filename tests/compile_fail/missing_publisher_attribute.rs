use ::yaaf::prelude::*;

#[derive(Clone, Debug)]
struct ValidMessage;

#[derive(Clone, Debug)]
struct InvalidMessage;

#[derive(Actor)]
#[handle(ValidMessage)]
struct MyActor;

#[async_trait]
impl Handler<ValidMessage> for MyActor {
    async fn handle(&mut self, ctx: &mut Context<Self>, _message: ValidMessage) {
        ctx.publish(InvalidMessage);
    }
}

fn main() {}
