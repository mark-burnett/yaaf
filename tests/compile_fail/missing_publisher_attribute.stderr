error[E0599]: no method named `publish` found for mutable reference `&mut yaaf::Context<MyActor>` in the current scope
  --> $DIR/missing_publisher_attribute.rs:16:13
   |
11 | struct MyActor;
   | --------------- doesn't satisfy `MyActor: Publisher<_>`
...
16 |         ctx.publish(InvalidMessage);
   |             ^^^^^^^ method not found in `&mut yaaf::Context<MyActor>`
   |
  ::: $WORKSPACE/src/context.rs
   |
   | pub struct Context<A> {
   | --------------------- doesn't satisfy `yaaf::Context<MyActor>: yaaf::PublishContext<_>`
   |
   = note: the method `publish` exists but the following trait bounds were not satisfied:
           `MyActor: Publisher<_>`
           which is required by `yaaf::Context<MyActor>: yaaf::PublishContext<_>`
